packages/myst-to-typst/src/math.ts

```diff
@@ -1,44 +1,57 @@
 import { texToTypst } from 'tex-to-typst';
-import type { Handler, ITypstSerializer } from './types.js';
+import type { Handler, ITypstSerializer, MathPlugins } from './types.js';
 
 function addMacrosToState(value: string, state: ITypstSerializer) {
   if (!state.options.math) return;
   Object.entries(state.options.math).forEach(([k, v]) => {
     const key = texToTypst(k);
-    if (value.includes(key)) state.data.mathPlugins[key] = texToTypst(v);
+    if (value.includes(key)) {
+      state.data.mathPlugins[key] = texToTypst(v.macro);
+    }
   });
 }
 
-type MathPlugins = ITypstSerializer['data']['mathPlugins'];
+function findCommandInMacro(macro: string, command: string) {
+  const escapedCommand = command.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+  const commandRe = new RegExp(`${escapedCommand}(?![a-zA-Z])`, 'g');
+  return [...macro.matchAll(commandRe)].length > 0;
+}
+
+function replaceCommandInMacro(macro: string, command: string, replaceValue: string) {
+  const escapedCommand = command.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+  const commandRe = new RegExp(`${escapedCommand}(?![a-zA-Z])`, 'g');
+  return macro.replaceAll(commandRe, replaceValue);
+}
 
 /**
- * Add any required recursive commands found, for example,
- * if only `\aMat` was included in the content, but sill requires other commands:
+ * MyST typst exports currently require recursive commands to be resolved
+ *
+ * As opposed to tex where recursive commands can remain and the latex compiler
+ * will handle them.
  *
- * ```javascript
- * {
- *    '\aNrm': "a",
- *    '\aMat': '[\mathrm{\aNrm}]',
- * }
- * ```
+ * All the state.options.math macros are passed to this function and resolved prior to
+ * exporting typst.
  */
-export function withRecursiveCommands(
-  state: ITypstSerializer,
-  plugins = state.data.mathPlugins,
-): MathPlugins {
-  if (!state.options.math) return plugins;
-  const pluginsList = Object.entries(plugins);
-  const addedPlugins: MathPlugins = {};
-  Object.entries(state.options.math).forEach(([k, v]) => {
-    const key = texToTypst(k);
-    if (plugins[key]) return;
-    pluginsList.forEach(([, value]) => {
-      if (value.includes(key)) addedPlugins[key] = texToTypst(v);
-    });
-  });
-  const newPlugins = { ...addedPlugins, ...plugins };
-  if (Object.keys(addedPlugins).length === 0) return newPlugins;
-  return withRecursiveCommands(state, newPlugins);
+export function resolveRecursiveCommands(plugins: MathPlugins): MathPlugins {
```

Comment:
The otehr way to go is to actually write these out to typst?

Reply:
In talking with @fwkoch this is dependent on the order, and that isn't something we want to figure out now.